src/libexpr/eval.cc

```diff
@@ -94,6 +94,16 @@ RootValue allocRootValue(Value * v)
 #endif
 }
 
+// Returns `true' is a string is a reserved keyword which requires quotation
+// when printing attribute set field names.
+static inline bool isReservedKeyword(const std::string_view str)
+{
+    static const std::unordered_set<std::string_view> reservedKeywords = {
+        "if", "then", "else", "assert", "with", "let", "in", "rec", "inherit",
+        "or", "throw", "import"
```

Comment:
```suggestion
        "throw", "import"
```

`or` is not a keyword. It does not need quoting in attribute names. The distinction between identifiers and attribute names is reflected in the function names in `print.cc`.

Reply:
Are you sure? It appears in https://github.com/NixOS/nix/blob/4539ab530ad23a8558512f784bd72c4cd0e72f13/src/libexpr/lexer.l#L138

I'll remove it but just wanted to sanity check.

Reply:
You're right, for whatever reason that's the only keyword from the lexer that works:
```
$ nix eval --expr '{ or = 1; }';
{ or = 1; }
```